src/main/java/frc/lib/io/motor/MotorTalonFX.java

```diff
@@ -0,0 +1,308 @@
+    private final StatusSignal<Angle> position;
+    private final StatusSignal<AngularVelocity> velocity;
+    private final StatusSignal<Voltage> supplyVoltage;
+    private final StatusSignal<Current> supplyCurrent;
+    private final StatusSignal<Current> torqueCurrent;
+    private final StatusSignal<Temperature> temperature;
+    private final StatusSignal<Double> closedLoopError;
+    private final StatusSignal<Double> closedLoopReference;
+    private final StatusSignal<Double> closedLoopReferenceSlope;
+
+    // Preconfigured control objects reused for efficiency
+    private final CoastOut coastControl = new CoastOut();
+    private final StaticBrake brakeControl = new StaticBrake();
+    private final Follower followControl = new Follower(0, false);
+    private final VoltageOut voltageControl = new VoltageOut(0).withEnableFOC(true);
+    private final TorqueCurrentFOC currentControl = new TorqueCurrentFOC(0);
+    private final DutyCycleOut dutyCycleControl = new DutyCycleOut(0).withEnableFOC(true);
+    private final DynamicMotionMagicTorqueCurrentFOC positionControl = new DynamicMotionMagicTorqueCurrentFOC(0, 0, 0,
+            0);
+    private final VelocityTorqueCurrentFOC velocityControl = new VelocityTorqueCurrentFOC(0);
+
+    // Executor for retrying config operations asynchronously
+    private BlockingQueue<Runnable> queue = new LinkedBlockingQueue<>();
+    private ThreadPoolExecutor threadPoolExecutor = new ThreadPoolExecutor(1, 1, 5,
+            java.util.concurrent.TimeUnit.MILLISECONDS, queue);
+
+    /**
+     * Attempts a config action up to 5 times until it succeeds.
+     *
+     * @param action The status-returning operation to retry.
+     */
+    public void checkErrorAndRetry(Supplier<StatusCode> action) {
+        threadPoolExecutor.submit(() -> {
+            for (int i = 0; i < 5; i++) {
+                StatusCode result = action.get();
+                if (result.isOK()) {
+                    break;
+                }
+            }
+        });
+    }
+
+    /**
+     * Constructs and initializes a TalonFX motor.
+     *
+     * @param id     CAN bus ID and bus name.
+     * @param config Configuration to apply to the motor.
+     */
+    public MotorTalonFX(CANDevice id, TalonFXConfiguration config) {
+        this.id = id;
+        motor = new TalonFX(id.id(), id.bus());
+
+        checkErrorAndRetry(() -> motor.getConfigurator().apply(config));
+
+        position = motor.getPosition();
+        velocity = motor.getVelocity();
+        supplyVoltage = motor.getSupplyVoltage();
+        supplyCurrent = motor.getSupplyCurrent();
+        torqueCurrent = motor.getTorqueCurrent();
+        temperature = motor.getDeviceTemp();
+        closedLoopError = motor.getClosedLoopError();
+        closedLoopReference = motor.getClosedLoopReference();
+        closedLoopReferenceSlope = motor.getClosedLoopReferenceSlope();
+
+        checkErrorAndRetry(() -> BaseStatusSignal.setUpdateFrequencyForAll(
```

Comment:
I think previously we had run into some issue with setting different update frequencies on the motors @skukshtel @MJWilk00 might recall